src/main/java/com/redhat/digkins/DiggerClient.java

```diff
@@ -38,19 +63,49 @@ public void createJob(String name, String gitRepo, String gitBranch) throws Digg
   }
 
   /**
-   * Create client using provided url and credentials
+   * Triggers a build for the given job and waits until it leaves the queue and actually starts.
+   * <p>
+   * Jenkins puts the build requests in a queue and once there is a slave available, it starts building
+   * it and a build number is assigned to the build.
+   * <p>
+   * This method will block until there is a build number, or the given timeout period is passed. If the build is still in the queue
+   * after the given timeout period, -1 is returned as the build number.
+   * <p>
+   * Please note that timeout period is never meant to be very precise. It has the resolution of {@link TriggerBuildService#POLL_PERIOD} because
+   * timeout is checked before every pull.
+   * <p>
+   * Similarly, -1 is returned if the build is stuck or cancelled on Jenkins side.
    *
-   * @param url      - jenkins url
-   * @param user     - jenkins user
-   * @param password - jenkins password
-   * @return client instance
+   * @param jobName name of the job to trigger the build
+   * @param timeout how many milliseconds should this call block before returning -1. Should be larger than {@link TriggerBuildService#FIRST_CHECK_DELAY}
+   * @return the build number. -1 if build is cancelled, stuck or in queue more than the given timeout period
+   * @throws DiggerClientException if connection problems occur during connecting to Jenkins
    */
-  public static DiggerClient from(String url, String user, String password) throws DiggerClientException {
+  public long build(String jobName, long timeout) throws DiggerClientException {
+    final TriggerBuildService triggerBuildService = new TriggerBuildService(jenkins);
     try {
-      JenkinsAuth jenkinsAuth = new JenkinsAuth(url, user, password);
-      return new DiggerClient(jenkinsAuth);
-    } catch (URISyntaxException e) {
-      throw new DiggerClientException("Invalid jenkins url format.");
+      return triggerBuildService.build(jobName, timeout);
+    } catch (IOException e) {
+      LOG.debug("Exception while connecting to Jenkins", e);
+      throw new DiggerClientException(e);
+    } catch (InterruptedException e) {
+      LOG.debug("Exception while sleeping between checks", e);
```

Comment:
Can we rephrase: "Exception while sleeping between checks" - It's not clear what that means for me.